b_cfn_elasticsearch_cloner/cloner_source/index.py

```diff
@@ -13,34 +14,55 @@
 try:
     from dynamodb_decode import DynamodbDecoder
 except ImportError as ex:
-    logger.exception("Failed import.")
+    logger.exception('Failed import.')
     from b_cfn_elasticsearch_cloner.cloner_source.dynamodb_decode import DynamodbDecoder
 
+ES_INDEX = os.environ['ES_INDEX_NAME']
+ES_ENDPOINT = os.environ['ES_DOMAIN_ENDPOINT']
+PRIMARY_KEY_FIELD = os.environ['PRIMARY_KEY_FIELD']
+REGION = os.environ['AWS_REGION']
 
-es_index = os.environ["ES_INDEX_NAME"]
-es_endpoint = os.environ["ES_DOMAIN_ENDPOINT"]
-primary_key_field = os.environ["PRIMARY_KEY_FIELD"]
-region = os.environ["AWS_REGION"]
+SAGEMAKER_ENDPOINT_NAME = os.environ.get('SAGEMAKER_ENDPOINT_NAME')
+SAGEMAKER_EMBEDDINGS_KEY = os.environ.get('SAGEMAKER_EMBEDDINGS_KEY')
+EMBEDDINGS_KEY = os.environ.get('EMBEDDINGS_KEY')
 
+sagemaker_client = boto3.client('sagemaker-runtime')
 
 credentials = boto3.Session().get_credentials()
 awsauth = AWS4Auth(
     credentials.access_key,
     credentials.secret_key,
-    region,
-    "es",
+    REGION,
+    'es',
     session_token=credentials.token,
 )
 
 es = Elasticsearch(
-    hosts=[{"host": es_endpoint, "port": 443}],
+    hosts=[{'host': ES_ENDPOINT, 'port': 443}],
     http_auth=awsauth,
     use_ssl=True,
     verify_certs=True,
     connection_class=RequestsHttpConnection,
 )
 
 
+def get_sm_features(source: str, sagemaker_client: Any, endpoint_name: str):
+    sagemaker_response = sagemaker_client.invoke_endpoint(
+        EndpointName=endpoint_name,
+        Body=source.encode(encoding='utf-8'),
+        ContentType='text/plain'
+    )
+    source_features = json.loads(sagemaker_response['Body'].read().decode())
+    return source_features
+
+
+def add_embedding(item: Dict) -> Dict:
+    source_features = get_sm_features(
+        item['question'], sagemaker_client=sagemaker_client, endpoint_name=SAGEMAKER_ENDPOINT_NAME)
+    item[EMBEDDINGS_KEY] = source_features[SAGEMAKER_EMBEDDINGS_KEY]
+    return item
+
+
 def handler(event: Dict[Any, Any], context: Any) -> None:
```

Comment:
Place entrypoint (handler) functions at the beggining of the file (after imports and configurations).